openhands/runtime/client/runtime.py

```diff
@@ -288,58 +288,64 @@ def _init_container(
             raise e
 
     @tenacity.retry(
-        stop=tenacity.stop_after_attempt(10),
-        wait=tenacity.wait_exponential(multiplier=2, min=1, max=20),
-        reraise=(ConnectionRefusedError,),
+        stop=tenacity.stop_after_attempt(5),
+        wait=tenacity.wait_exponential(multiplier=2, min=2, max=20),
+        retry=tenacity.retry_if_exception_type(ConnectionRefusedError),
+        reraise=True,  # Re-raise exceptions after retries
+        retry_error_callback=lambda retry_state: None,
```

Comment:
If I'm reading this right, now we retry only if it's ConnectionRefusedError. Is that the only one we use?

Reply:
this is what it was before:
```
    @tenacity.retry(
        stop=tenacity.stop_after_attempt(10),
        wait=tenacity.wait_exponential(multiplier=2, min=1, max=20),
        reraise=(ConnectionRefusedError,),
    )
```
Did Sonnet do it wrong? :P

Reply:
Probably could revert to how it was before...

Reply:
The revert might be good, I was just seeing some other exception in `_wait_until_alive`. Thanks, let me give it a quick run, this is so weird to reproduce but a few tries won't hurt. 😅